onnxscript/ir/passes/_pass_infra.py

```diff
@@ -97,76 +145,102 @@ def ensures(self, model: ir.Model) -> None:
         del model  # Unused
 
 
-class PassManager:
+class InPlacePass(PassBase):
+    """A pass that modifies the input model in place and returns it."""
+
+    @property
+    def in_place(self) -> bool:
+        return True
+
+    @property
+    def changes_input(self) -> bool:
+        return True
+
+
+class OutOfPlacePass(PassBase):
```

Comment:
nit: I wonder if `FunctionalPass` would be a better name ... something that conveys the does-not-modify-input nature of the pass

Reply:
Sure!

Reply:
Done